lib/Service/ParticipantService.php

```diff
@@ -300,6 +307,66 @@ public function addUsers(Room $room, array $participants): void {
 		$this->dispatcher->dispatch(Room::EVENT_AFTER_USERS_ADD, $event);
 	}
 
+	/**
+	 * @param Room $room
+	 * @param IGroup $group
+	 * @param Participant[] $existingParticipants
+	 */
+	public function addGroup(Room $room, IGroup $group, array $existingParticipants = []): void {
```

Comment:
As adding a group is not a frequent action, maybe it would be better to always fetch the participants inside the method even if that duplicates the query to ensure that the method is not misused in some future code with a wrong list of participants?

Reply:
yeah thought so too, but well just leave it empty then? The thing is if you ever add a group to a 2k users room it will load and create the 2k objects twice... not sure how longthis will scale